src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdApi.java

```diff
@@ -59,7 +59,31 @@ public CcdApi(
 
     private CaseDetails retrieveCase(String caseRef, String jurisdiction) {
         CcdAuthenticator authenticator = authenticatorFactory.createForJurisdiction(jurisdiction);
-        return feignCcdApi.getCase(authenticator.getUserToken(), authenticator.getServiceToken(), caseRef);
+        try {
+            return feignCcdApi
+                .getCase(authenticator.getUserToken(), authenticator.getServiceToken(), caseRef);
+        } catch (FeignException ex) {
+            removeFromIdamCache(ex, jurisdiction);
+            throw ex;
+        }
+    }
+
+    private SearchResult searchCases(
+        String jurisdiction,
+        String caseType,
+        String searchString) {
+        CcdAuthenticator authenticator = authenticatorFactory.createForJurisdiction(jurisdiction);
+        try {
+            return feignCcdApi.searchCases(
+                authenticator.getUserToken(),
+                authenticator.getServiceToken(),
+                caseType,
+                searchString
+            );
+        } catch (FeignException ex) {
+            removeFromIdamCache(ex, jurisdiction);
+            throw ex;
+        }
```

Comment:
Same as above

Reply:
it removes when it is 401 or 403. inside the method `removeFromIdamCache`